### ISSUE TITLE
Problems with the launch.json system

### ISSUE THREAD

**Issue by carlca** [Microsoft/vscode-go#1403]
Hello, again, I don't know whether this problem is peculiar to VSCode-go but I'm having problems understanding and getting my `launch.json` mechanism under control.

![screen shot 2017-12-12 at 20 45 07](https://user-images.githubusercontent.com/260612/33907717-018fff6e-df7e-11e7-96dc-99a8cdb461c4.png)
 
In the attached screenshot we can see the dropdown with just one `launch.json` available to run, namely `gotwitter`. Yet, the `launch.json` that I want to run, the one being edited, is not available for running or is it in the list of ones to be added. That `launch.json` is in a separate folder called `sqlservertest` and is entirely separate from the `sqlserver` folder. How do I get it to be associated with the `sqlservertest.go` file and run when selected from the Debug dropdown?

**Comment by carlca** [comment]
In the replies to https://github.com/Microsoft/vscode-cpptools/issues/396 it says how there is supposed to be a "Configure or Fix launch.json button in the Debug pane". Where is this button. I cannot find it?

**Comment by ramya-rao-a** [comment]
Are you in a multi folder set up? And each of those `Add Config ()` correspond to one of the folders?

**Comment by ramya-rao-a** [comment]
Please share the folder structure as well. Is `sqlservertest` one of the folders added to the workspace? Or is it a folder under another folder that was added to the workspace?

**Comment by carlca** [comment]
![screen shot 2017-12-13 at 00 06 12](https://user-images.githubusercontent.com/260612/33915114-82d89034-df99-11e7-95a9-c709a950c925.png)

Hi Ramya, that is what my folder structure looks like at the moment. 

The drop-down list is missing some `Add Config() s` but I notice that they are subfolders of `BigData`. That includes `sqlservertest`.

I've tried saving my Workspace and although I got an error message the first time, it seems to have worked the second time.

Am I right in thinking that the only folders that contribute to the list of `Add Config() s` are top-level folders in the Workspace? That may have been the crucial missing part of my understanding!

**Comment by ramya-rao-a** [comment]
> Am I right in thinking that the only folders that contribute to the list of Add Config() s are top-level folders in the Workspace?

Yes, that's right.

**Comment by carlca** [comment]
OK I've added two folders which are already sub-folders, to be immediate folders within the workspace. I've just tried to resave the workspace. The error I got before has reappeared:

![screen shot 2017-12-13 at 00 27 24](https://user-images.githubusercontent.com/260612/33915718-b33fc7ee-df9c-11e7-94b3-9205c0e88e92.png)

Never mind that. I've got access to the correct `launch.json` configs now. Thanks so much, Ramya!!
